pkg/authn/oidc.go

```diff
@@ -36,20 +37,30 @@ var (
 )
 
 // NewOIDCAuthenticator returns OIDC authenticator
-func NewOIDCAuthenticator(config *OIDCConfig) (*OIDCAuthenticator, error) {
+func NewOIDCAuthenticator(ctx context.Context, config *OIDCConfig) (*OIDCAuthenticator, error) {
 	dyCA, err := dynamiccertificates.NewDynamicCAContentFromFile("oidc-ca", config.CAFile)
 	if err != nil {
 		return nil, err
 	}
 
-	tokenAuthenticator, err := oidc.New(oidc.Options{
-		IssuerURL:            config.IssuerURL,
-		ClientID:             config.ClientID,
+	tokenAuthenticator, err := oidc.New(ctx, oidc.Options{
+		JWTAuthenticator: apiserver.JWTAuthenticator{
```

Comment:
can we remove the old options and just wire the new config, since we have the chance?

Reply:
@stlaz, WDYT?

https://github.com/brancz/kube-rbac-proxy/pull/300/commits/a610979bafbaa50c79e334ebe3dc92aa7e77ca2f

Reply:
I understand. Let me remove the change.

Introducing the suggested change, is definitely another PR.

Reply:
would you create a k/k issue and link it to this PR, please?

Reply:
What do you expect me to create an issue for? That we finally need to move that into one place instead of copying it across different places and that this PR is an example of why we need it?